src/Grb.Building.Processor.Upload/Zip/Validators/ZipArchiveValidator.cs

```diff
@@ -28,79 +29,104 @@ public ZipArchiveProblems Validate(ZipArchive archive)
-                            _validators.Keys,
-                            StringComparer.InvariantCultureIgnoreCase
-                        )
-                    );
-
-                    foreach (var file in requiredFiles.OrderBy(file =>
-                                 Array.IndexOf(ValidationOrder, file.ToUpperInvariant())))
+                    )
+                );
+
+                foreach (var file in requiredFiles
+                             .OrderBy(file => Array.IndexOf(ValidationOrder, file.ToUpperInvariant())))
+                {
+                    if (!_validators.TryGetValue(file, out var validator))
                     {
-                        if (_validators.TryGetValue(file, out var validator))
+                        continue;
+                    }
+
+                    var problemsInFile = validator.Validate(archive.GetEntry(file));
+                    var problemsInFileByErrorType = problemsInFile
+                        .SelectMany(kvp =>
                         {
-                            var fileProblems = validator.Validate(archive.GetEntry(file));
-                            var fileProblemsGrouped = fileProblems
-                                .SelectMany(kvp =>
-                                    kvp.Value.Select(errorType =>
-                                        new { ErrorType = errorType, RecordNumber = kvp.Key }))
-                                .GroupBy(x => x.ErrorType)
-                                .ToDictionary(group => group.Key, group => group.Select(x => x.RecordNumber).ToList());
-
-                            foreach (var kvp in fileProblemsGrouped)
+                            var (recordNumber, errorTypes) = kvp;
+                            return errorTypes.Select(errorType => new
                             {
-                                problems.Add(new FileError(file, kvp.Key.ToString(),
-                                    new ProblemParameter("recordnumbers", string.Join(",", kvp.Value))));
-                            }
-                        }
+                                ErrorType = errorType,
+                                RecordNumber = recordNumber
+                            });
+                        })
+                        .GroupBy(x => x.ErrorType)
+                        .ToDictionary(
+                            group => group.Key,
+                            group => group.Select(x => x.RecordNumber).ToList());
+
+                    foreach (var problemsForErrorType in problemsInFileByErrorType)
+                    {
+                        problems.Add(new FileError(
+                            file,
+                            // I think we should translate below to a proper string/text as this will be showed in the job ticket error result.
+                            problemsForErrorType.Key.ToString(),
+                            new ProblemParameter("recordNumbers", string.Join(",", problemsForErrorType.Value))));
```

Comment:
Here we are also creating one ticket error for multiple record numbers, because they are of the same error type. In the job processor, we will create a ticket error per error returned by the backoffice API or lambda regardless if there are multiple errors of the same "type".